### DISCUSSION TITLE
Does the v2.0 beta version have a chainChanged event and how to call it?

### DISCUSSION THREAD

**Discussion by 1648045459** [WalletConnect/walletconnect-monorepo#1119] [Developer Support] [answered]
Does the v2.0 beta version have a chainChanged event and how to call it?

**Comment by hackyguru** [comment, marked as answer]
v2 is built in a way that allows any method / event to be emitted.
To have chainChanged event, you'd need to have these requirements satisfied :

1. When dapp / wallet establish a session, chainChanged event should be included in pairing namespace.
2. wallet / dapp can then call .emit method and pass all necessary data, for that event.

Hope this helps!

**Comment by 1648045459** [comment]


**Reply by Mukesh6201** [reply]
@hackyguru can you tell me how to use wallet connect v2.0 library there is no documentation available over internet. I am working with kotlin . Also can you refer me to any sample app based on v2.0 if available?

**Comment by hackyguru** [comment]
@Mukesh6201 The v2 docs are available at https://docs.walletconnect.com/2.0/ . However, some of the portions might be incomplete for now but we are working on publishing more content on v2 docs in the coming weeks!
